utils/src/main/java/software/amazon/awssdk/utils/async/ByteBufferStoringSubscriber.java

```diff
@@ -56,6 +57,10 @@ public class ByteBufferStoringSubscriber implements Subscriber<ByteBuffer> {
 
     private final Phaser phaser = new Phaser(1);
 
+    private final AtomicInteger outstandingDemand = new AtomicInteger(0);
+
+    private final Optional<Integer> maximumOutstandingDemand;
```

Comment:
minor: per our guideline, Optional SHOULD NOT be used for member variables

https://github.com/aws/aws-sdk-java-v2/blob/master/docs/guidelines/UseOfOptional.md